controls/raddocking/how-to/add-menu-items-to-the-radpanes-menu.md

```diff
@@ -14,136 +14,220 @@ The purpose of this tutorial is to show you how to customize the __RadPane__ men
       
 >tipFor more information about the __RadPane__ structure, read [here]({%slug raddocking-panes-radpane%}) or read [this one]({%slug raddocking-features-panes-panes-menu%}) if you want to learn more about the __RadPane's__ menu.
 
-## Customizing the RadPane's Menu
+## Customizing the RadPane's Menu 
 
 In order to add custom commands to the __RadPane's__ menu, you should perform the following steps:
 
-1. Create a singleton property of type __Telerik.Windows.Controls.RoutedUICommand__. In the following example the custom command is named __CloseAllPanesButThis__.
+1. You can create a custom class CustomRadDockingCommands. Inside you can create a singleton property of type __Telerik.Windows.Controls.RoutedUICommand__. __Example 1__ demonstrate how you can do that.
 	
-	#### __C#__
-
+	#### __[C#] Example 1: Create Singleton command property__
 	{{region cs-raddocking-how-to-add-menu-items-to-the-radpanes-menu_0}}
-		public static class RadDockingCommands
+		public class CustomRadDockingCommands
 		{
-		    private static RoutedUICommand closeAllPanesButThisCommand;
-		
-		    public static RoutedUICommand CloseAllPanesButThisCommand
-		    {
-		        get
-		        {
-		            if (closeAllPanesButThisCommand == null)
-		            {
-		                closeAllPanesButThisCommand = new RoutedUICommand("Close all panes but this", "CloseAllPanesButThisCommand", typeof(RadDockingCommands));
-		            }
-		            return closeAllPanesButThisCommand;
-		        }
-		    }
+			public static class RadDockingCommands
+			{
+				private static RoutedUICommand closeAllPanesButThisCommand;
+			
+				public static RoutedUICommand CloseAllPanesButThisCommand
+				{
+					get
+					{
+						if (closeAllPanesButThisCommand == null)
+						{
+							closeAllPanesButThisCommand = new RoutedUICommand("Close all panes but this!", "CloseAllPanesButThisCommand", typeof(RadDockingCommands));
+						}
+						return closeAllPanesButThisCommand;
+					}
+				}
+			}
 		}
 	{{endregion}}
 
-1. Create a custom __DataTemplate__ by using the Telerik __RadContextMenu__ and __RadMenuItem__ controls. You can use either the built-in __RadDockingCommands__ or create a custom one - the both approaches are demonstrated in this topic. Set the created __DataTemplate__ to the __RadPane__'s __ContextMenuTemplate__ property. In the following code-snippet is shown how to set the __RadPane__'s __ContextMenuTemplate__ property.
+1. Create a custom __DataTemplate__ by using the Telerik __RadContextMenu__ and __RadMenuItem__ controls. You can use either the built-in __RadDockingCommands__ or create a custom one - the both approaches are demonstrated in this topic. Set the created __DataTemplate__ to the __RadPane__'s __ContextMenuTemplate__ property. __Example 2__ show how to set the __RadPane__'s __ContextMenuTemplate__ property.

```

Comment:
Telerik's RadContextMenu ..  
both approaches (without the)..
to RadPane's ContexteMenuTemplate (without the)..
RadPane's ContextMenuTemplate(without the)..